File: NekrasovskyAPP/NekrasovskyAPP/Platforms/Android/AlarmSoundService.cs
using Android.Content;
using Android.Media;

namespace NekrasovskyAPP.Services
{
    public partial class AlarmSoundService
    {
        private MediaPlayer? _mediaPlayer;
        private ToneGenerator? _toneGenerator;
        private int? _previousAlarmVolume;
        private int? _previousMusicVolume;

        partial void StartPlatformAlarm(CancellationToken token)
        {
            try
            {
                var context = Android.App.Application.Context;
                if (context == null) return;

                var audioManager = context.GetSystemService(Context.AudioService) as AudioManager;
                if (audioManager != null)
                {
                    _previousAlarmVolume = audioManager.GetStreamVolume(Android.Media.Stream.Alarm);
                    _previousMusicVolume = audioManager.GetStreamVolume(Android.Media.Stream.Music);

                    var maxAlarmVolume = audioManager.GetStreamMaxVolume(Android.Media.Stream.Alarm);
                    var maxMusicVolume = audioManager.GetStreamMaxVolume(Android.Media.Stream.Music);

                    audioManager.SetStreamVolume(Android.Media.Stream.Alarm, maxAlarmVolume, VolumeNotificationFlags.RemoveSoundAndVibrate);
                    audioManager.SetStreamVolume(Android.Media.Stream.Music, maxMusicVolume, VolumeNotificationFlags.RemoveSoundAndVibrate);
                }

                // Пробуем загрузить MP3
                try
                {
                    var assets = context.Assets;
                    if (assets != null)
                    {
                        using var assetFd = assets.OpenFd("alarm.mp3");
                        _mediaPlayer = new MediaPlayer();
                        _mediaPlayer.SetDataSource(assetFd.FileDescriptor, assetFd.StartOffset, assetFd.Length);
                        _mediaPlayer.Looping = true;
                        _mediaPlayer.Prepare();
                        _mediaPlayer.Start();
                        return;
                    }
                }
                catch
                {
                    // MP3 не найден - используем ToneGenerator
                }

                // Fallback: ToneGenerator
                _toneGenerator = new ToneGenerator(Android.Media.Stream.Alarm, 100);

                Task.Run(async () =>
                {
                    try
                    {
                        while (!token.IsCancellationRequested)
                        {
                            _toneGenerator?.StartTone(Tone.CdmaEmergencyRingback, 1500);
                            await Task.Delay(2000, token);
                        }
                    }
                    catch (OperationCanceledException) { }
                }, token);
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine($"Android alarm error: {ex.Message}");
            }
        }

        partial void StopPlatformAlarm()
        {
            try
            {
                var context = Android.App.Application.Context;
                var audioManager = context?.GetSystemService(Context.AudioService) as AudioManager;
                if (audioManager != null)
                {
                    if (_previousAlarmVolume.HasValue)
                    {
                        audioManager.SetStreamVolume(Android.Media.Stream.Alarm, _previousAlarmVolume.Value, VolumeNotificationFlags.RemoveSoundAndVibrate);
                        _previousAlarmVolume = null;
                    }

                    if (_previousMusicVolume.HasValue)
                    {
                        audioManager.SetStreamVolume(Android.Media.Stream.Music, _previousMusicVolume.Value, VolumeNotificationFlags.RemoveSoundAndVibrate);
                        _previousMusicVolume = null;
                    }
                }

                _toneGenerator?.StopTone();
                _toneGenerator?.Release();
                _toneGenerator = null;

                _mediaPlayer?.Stop();
                _mediaPlayer?.Release();
                _mediaPlayer?.Dispose();
                _mediaPlayer = null;
            }
            catch { }
        }
    }
}

File: android-app/app/src/main/java/com/nekrasovsky/android/data/sound/AlarmSoundService.kt
package com.nekrasovsky.android.data.sound

import android.content.Context
import android.media.AudioManager
import android.media.MediaPlayer
import android.media.ToneGenerator
import android.util.Log

class AlarmSoundService(private val context: Context) {
    private var mediaPlayer: MediaPlayer? = null
    private val toneGenerator = ToneGenerator(AudioManager.STREAM_NOTIFICATION, 100)

    fun playAlarmSound() {
        try {
            // Пытаемся использовать системный звук тревоги
            val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
            
            // Используем ToneGenerator для воспроизведения тонального сигнала тревоги
            toneGenerator.startTone(ToneGenerator.TONE_CDMA_ALERT_CALL_GUARD, 2000)
            
            Log.d("AlarmSoundService", "Звук тревоги воспроизведен")
        } catch (e: Exception) {
            Log.e("AlarmSoundService", "Ошибка воспроизведения звука", e)
        }
    }

    fun stopAlarmSound() {
        try {
            toneGenerator.stopTone()
            mediaPlayer?.release()
            mediaPlayer = null
        } catch (e: Exception) {
            Log.e("AlarmSoundService", "Ошибка остановки звука", e)
        }
    }
}
